File: core/network/analytics/src/main/java/com/appcoins/wallet/core/network/analytics/AnalyticsAPI.java
package com.appcoins.wallet.core.network.analytics;

import cm.aptoide.analytics.AnalyticsManager.Action;
import io.reactivex.Completable;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface AnalyticsAPI {

  @POST("user/addEvent/action={action}/context=WALLET/name={name}") Completable registerEvent(
      @Path("action") Action action, @Path("name") String eventName, @Body AnalyticsBody body);
}

File: core/network/analytics/src/main/java/com/appcoins/wallet/core/network/analytics/api/AnalyticsApi.kt
package com.appcoins.wallet.core.network.analytics.api

import cm.aptoide.analytics.AnalyticsManager
import com.appcoins.wallet.core.network.analytics.AnalyticsBody
import io.reactivex.Completable
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Path

interface AnalyticsApi {
  @POST("user/addEvent/action={action}/context=WALLET/name={name}")
  fun registerEvent(
    @Path("action") action: AnalyticsManager.Action?,
    @Path("name") eventName: String?,
    @Body body: AnalyticsBody?
  ): Completable
}